Add configuration example with `PHP` format

GromNaN opened the pull request:
Fix #806

franmomu committed:
Remove bundle examples

GromNaN committed:
Add PHP config examples

malarzm commented on docs/config.rst:
```diff
@@ -214,6 +258,34 @@ The following configuration shows a bunch of mapping examples:
+                    ->type('xml')
+                ->mapping('MyBundle3')
+                    ->type('attribute')
+                    ->dir(__DIR__.'/../src/Document')
```
I think the directories should be consistent with previous sections, so this should read `Documents/`

GromNaN replied:
Using the constants is one of the advantages of using the PHP configuration, I think it's better to show the full feature as it should be used in the documentation.

malarzm commented on docs/config.rst:
```diff
@@ -214,6 +258,34 @@ The following configuration shows a bunch of mapping examples:
+                    ->dir(__DIR__.'/../src/Document')
+                ->mapping('MyBundle4')
+                    ->type('xml')
+                    ->dir(__DIR__.'/../config/doctrine/mapping')
```
This is missing Resources. Or if `Resources` is no longer encouraged we should update previous examples

malarzm commented on docs/config.rst:
```diff
@@ -214,6 +258,34 @@ The following configuration shows a bunch of mapping examples:
+                ->mapping('MyBundle5')
+                    ->type('xml')
+                    ->dir('my-bundle-mappings-dir')
+                    ->alias('BundleAlias')
```
Haven't we removed aliases and entire `Smh:Document` notation along with doctrine/common 3.0 requirement? Or is this something else?

malarzm commented on docs/config.rst:
```diff
@@ -214,6 +258,34 @@ The following configuration shows a bunch of mapping examples:
+                    ->alias('BundleAlias')
+                ->mapping('MyBundle6')
+                    ->type('xml')
+                    ->dir('%kernel.project_dir%/src/vendor/DoctrineExtensions/lib/DoctrineExtensions/Documents')
```
I see that Symfony provides a `param` function that can be used instead `%notation%`. Shall we endorse the new way or is there no community consensus yet?

GromNaN replied:
Yes, I'll use it.

malarzm commented on docs/config.rst:
```diff
@@ -298,6 +381,24 @@ Filters may be registered with a document manager by using the following syntax:
+        return static function (DoctrineMongodbConfig $config): void {
+            $config->documentManager('default')
+                ->filter('basic_filter')
+                    ->class('Vendor\Filter\BasicFilter')
```
We should be encouraging `::class` notation in both `->class` calls

malarzm commented on docs/config.rst:
```diff
@@ -684,6 +863,69 @@ Full Default Configuration
+            $config->hydratorNamespace('Hydrators');
+            $config->proxyDir('%kernel.cache_dir%/doctrine/odm/mongodb/Proxies');
+            $config->proxyNamespace('Proxies');
+            $config->fixtureLoader('Symfony\Bridge\Doctrine\DataFixtures\ContainerAwareLoader');
```
`::class` please, everywhere

malarzm reviewed (commented):
Sorry for the number of comments 🙈

GromNaN commented on docs/config.rst:
```diff
@@ -684,6 +863,69 @@ Full Default Configuration
+                    ->type('')
+                    ->class('')
+                    ->host('')
+                    ->port('')
```
This list of "defaults" is incorrect. There is no default values as this are not set by default.

GromNaN committed:
Review

franmomu reviewed (commented):
I remembered that I starting changing references to bundle to fix https://github.com/doctrine/DoctrineMongoDBBundle/issues/536, we should remove them here as well (I've just created https://github.com/doctrine/DoctrineMongoDBBundle/pull/835)

GromNaN committed:
Apply Bundle structure removal

GromNaN committed:
Replace sample configuration from flex recipes examples

https://github.com/symfony/recipes-contrib/blob/main/doctrine/mongodb-odm-bundle/4.4/config/packages/doctrine_mongodb.yaml

GromNaN commented:
@franmomu @malarzm The PR have been updated after your reviews.
I updated the first example to take the default configuration provided in the [Symfony recipe.](https://github.com/symfony/recipes-contrib/blob/main/doctrine/mongodb-odm-bundle/4.4/config/packages/doctrine_mongodb.yaml). ~The `is_bundle` part is very important since Bundle structure is no longer the default.~

GromNaN commented on docs/config.rst:
```diff
@@ -18,11 +18,11 @@ Sample Configuration
-                            class: Class\Example\Filter\ODM\ExampleFilter
-                            enabled: true
+                        App:
+                            is_bundle: false
```
The `is_bundle` and `alias` configs are not necessary for a default setup. I update the recipe: https://github.com/symfony/recipes-contrib/pull/1582

GromNaN commented on src/DependencyInjection/DoctrineMongoDBExtension.php:
```diff
@@ -462,14 +462,14 @@ private function normalizeDriverOptions(array $connection): array
-     *         MyBundle3: { type: attribute }
-     *         MyBundle4: { type: xml, dir: Resources/config/doctrine/mapping }
-     *         MyBundle5:
+     *         App1: ~
```
This configs need to be updated, as `dir` is required when outside of a bundle: https://github.com/symfony/symfony/blob/7bbc9be7ea7d607fa901edb2068c2cea13d24978/src/Symfony/Bridge/Doctrine/DependencyInjection/AbstractDoctrineExtension.php#L68

franmomu replied:
I was wondering if we should keep these examples here if we have them in the documentation.

Maybe we can remove them here and update the ones in the documentation adding the `dir` option?

GromNaN replied:
Having an example for attribute and xml, and for a bundle would be enough.

malarzm commented on docs/config.rst:
```diff
@@ -138,9 +182,9 @@ can control. The following configuration options exist for a mapping:
+  for an application called ``App``, the prefix would be
+  ``App\Document``.
 
 - ``alias`` Doctrine offers a way to alias document namespaces to simpler,
```
I still think we've missed deprecating this one :)

GromNaN replied:
Let's deprecate it for 5.1 and remove it from the doc.

malarzm reviewed (approved):
Thanks for the changes! 🍻

GromNaN commented:
Status: I need to check the examples. I'm not sure everything is correct and optimal.
